electron/src/main.ts

```diff
@@ -283,6 +304,48 @@ const handleAppEvents = () => {
     isQuitting = true;
   });
 
+  app.on('login', async (event, webContents, request, authInfo, callback) => {
+    if (authInfo.isProxy) {
+      event.preventDefault();
+
+      if (authenticatedProxyInfo && authenticatedProxyInfo.username && authenticatedProxyInfo.password) {
+        logger.info('Sending provided credentials to authenticated proxy ...');
+        return callback(authenticatedProxyInfo.username, authenticatedProxyInfo.password);
+      }
+
+      const systemProxy = await getProxySettings();
+      const systemProxySettings = systemProxy && (systemProxy.http || systemProxy.https);
+
+      if (systemProxySettings) {
+        authenticatedProxyInfo = new URL(
+          `http://${systemProxySettings.credentials.username}:${systemProxySettings.credentials.password}@${authInfo.host}:${authInfo.port}`,
+        );
+        return callback(systemProxySettings.credentials.username, systemProxySettings.credentials.password);
+      }
+
+      ipcMain.once(
+        EVENT_TYPE.PROXY_PROMPT.SUBMITTED,
+        (event: IpcMessageEvent, data: {password: string; username: string}) => {
+          logger.log('Proxy prompt was submitted');
+          authenticatedProxyInfo = new URL(
+            `http://${data.username}:${data.password}@${authInfo.host}:${authInfo.port}`,
```

Comment:
Maybe we should ask and receive the type of proxy here as well as we seems to force the use of HTTP

Reply:
If `authInfo.port` is not available we should rely on the protocol defaults:

```suggestion
            `http://${data.username}:${data.password}@${authInfo.host}${authInfo.port ? `:${authInfo.port}` : ''}`,
```

Reply:
I figured out that `authInfo.host` already includes the port.